src/api/app/models/workflow.rb

```diff
@@ -74,10 +74,17 @@ def supported_filters
 
   def event_matches_event_filter?
     return true unless supported_filters.key?(:event)
-    return true if filters[:event] == 'push' && scm_webhook.push_event?
-    return true if filters[:event] == 'pull_request' && scm_webhook.pull_request_event?
 
-    false
+    case filters[:event]
```

Comment:
I noticed that there was a typo here since the last approval from @krauselukas.

I changed this line from `case filters[:event]` to `case filters[:events]`.